### DISCUSSION TITLE
Assign IPAddress to a Virtual interface issue

### DISCUSSION THREAD

**Discussion by evsoares-meli** [netbox-community/netbox#7047] [Q&A] [answered]
Hello guys, I'm having trouble assigning an IP to a virtual interface via custom script and/or nbshell, the code returns this error: **TypeError: Direct assignment to the reverse side of a related set is prohibited. Use interface.set() instead.**

This is the code: (netbox v2.10.8)
```#set up mgmt IP
iface = Interface.objects.get (device = fw, name = 'dmz')

fw_mgmt_ip = IPAddress (
    address = '10.0.20.10/28',
    interface = iface
)

fw_mgmt_ip.save ()

fw.primary_ip4 = fw_mgmt_ip
fw.save()
return fw
```

Sorry for my bad understand of django.
Thanks in advance

**Comment by candlerb** [comment, marked as answer]
As of [Netbox v2.9.0](https://github.com/netbox-community/netbox/releases/v2.9.0), Interface was split into `Interface` and `VMInterface`.

Hence there is no longer an `interface` attribute on IPAddress, there is `assigned_object`.  This is because an IPAddress can be linked to either an Interface or a VMInterface, and it uses a generic relation for this.

Here is some working code:
https://github.com/netbox-community/reports/blob/master/scripts/create_vm.py#L88

It creates a VirtualMachine, a VMInterface and an IPAddress, and links them together.
